internal/controllers/clusterextension_controller.go

```diff
@@ -314,10 +314,16 @@ func SetDeprecationStatus(ext *ocv1alpha1.ClusterExtension, bundleName string, d
 			case declcfg.SchemaPackage:
 				deprecations[ocv1alpha1.TypePackageDeprecated] = entry
 			case declcfg.SchemaChannel:
-				if ext.Spec.Channel != entry.Reference.Name {
-					continue
+				// TODO: How should we handle doing channel deprecation with
+				// multiple channels being specified?
+				// For now, if it is possible it is from a deprecated channel
+				// set it.
```

Comment:
I wasn't sure how we should approach this so I went with this for now. I do wonder if we should even support "channel deprecation" when looking at multiple channels for filtering.

Reply:
I guess it matters if you care about "any" vs. "all". If you have to check for "all", then you might need a new map (or similar) to track the deprecation status for each channel.